transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/persistence/SchemaManager.java

```diff
@@ -55,21 +122,47 @@ private void ensureConfigured(SQLiteDatabase db) {
   @Override
   public void onCreate(SQLiteDatabase db) {
     ensureConfigured(db);
-    databaseBootstrapClient.bootstrap(db);
+    upgrade(db, 0, schemaVersion);
   }
 
   @Override
   public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
     ensureConfigured(db);
+    upgrade(db, oldVersion, newVersion);
   }
 
   @Override
   public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion) {
-    ensureConfigured(db);
+    db.execSQL(DROP_EVENTS_SQL);
+    db.execSQL(DROP_EVENT_METADATA_SQL);
+    db.execSQL(DROP_CONTEXTS_SQL);
+    // Indices are dropped automatically when the tables are dropped
+
+    onCreate(db);
```

Comment:
I think this is wrong, and should instead be `onUpgrade(0, newVersion)`, am I missing something?

Reply:
Net effect is the same.

```
@Override
  public void onCreate(SQLiteDatabase db) {
    ensureConfigured(db);
    // schemaVersion here should be the same as the newVersion in onDowngrade.
    upgrade(db, 0, schemaVersion);
  }
```